docs/src/modules/reference/pages/migration-guide.adoc

```diff
@@ -1,3 +1,25 @@
 = Migration Guide
 
 include::ROOT:partial$include.adoc[]
+
+== SDK 3.3.0
+
+This release runs all user component and endpoint logic on Java 21 virtual threads, this means it is
+safe to do IO calls without blocking the underlying OS thread.
+
+This means it is no longer any need to use `CompletionStage` and `CompletableFuture` composition when
+doing component client, HTTP client and gRPC client calls. Where a service would previously call `.invokeAsync()` and get a future back, you can now instead use `.invoke()` and directly operate on or return the return value.
+
+=== Breaking changes
+
+To make the SDK as easy to use as possible we have made the following breaking changes, you will have
+to update your service if using any of the following features:
+
+ * Scheduling and cancelling timers using `TimerScheduler.startSingleTimer(...)` and `TimerScheduler.cancel()` are now blocking and returns `void`. If you still prefer to use future composition, new methods called `startSingleTimerAsync` and `cancelAsync` has been added.
```

Comment:
```suggestion
 * Scheduling and cancelling timers using `TimerScheduler.startSingleTimer(...)` and `TimerScheduler.cancel()` are now blocking and returns `void`. If you still prefer to use future composition, new methods called `startSingleTimerAsync` and `cancelAsync` have been added.
```

Reply:
I know that this was already discussed and decided to break the API. 
  
It was suggested that we could add a `createSingleTimer`, why did we give up?

We can introduce `createSingleTimer` and `createSingleTimerAsync` and deprecate `startSingleTimer`. 

In restrospect, I think the original name is not great. Create a timer